### PR TITLE
Feature/improve metrics initialisation

### PR DESCRIPTION
**Pull request by solidgoldpig**

### CODE REVIEW

File: lib/client/metrics/metrics.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  })
+
+  metricsClient = {
+    apiMetrics,
+    requestMetrics
+  }
+}
+
+const getMetricsClient = () => {
+  init()
```
Comment (umaar) [inline lib/client/metrics/metrics.js]: Wonder if there are any side effects of putting this `init` within a getter

Comment (umaar) [inline lib/client/metrics/metrics.js]: Wonder if there are any side effects of putting this init within a getter

File: package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "lint:lib": "eslint lib",
     "test": "npm-run-all lint test:all",
     "test:all": "npm-run-all test:unit",
-    "test:unit": "multi-tape lib/**/*.unit.spec.js lib/**/**/*.unit.spec.js lib/middleware/**/*.unit.spec.js lib/page/**/*.unit.spec.js lib/**/**/**/*.unit.spec.js"
+    "test:unit": "USER_DATASTORE_URL='' USER_FILESTORE_URL='' SUBMITTER_URL='' SERVICE_SECRET='' SERVICE_TOKEN='' SERVICE_SLUG='' multi-tape lib/**/*.unit.spec.js lib/**/**/*.unit.spec.js lib/middleware/**/*.unit.spec.js lib/page/**/*.unit.spec.js lib/**/**/**/*.unit.spec.js"
```
Comment (umaar) [inline package.json]: Could this cause probs when running tests individually? e.g. when working on a test I normally do `./node_modules/.bin/tape lib/client/metrics/metrics.unit.spec.js` - maybe the `test.only()` is better though

File: lib/client/metrics/metrics.js (hunk 1 above)
Comment (solidgoldpig) [reply lib/client/metrics/metrics.js]: LIke what?

File: package.json (hunk 2 above)
Comment (solidgoldpig) [reply package.json]: Caveat emptor/tester

If you're running the tests individually you'll see the error clearly for that one file and it's up to you for your ENV to be in the correct state ;)

File: lib/client/metrics/metrics.js (hunk 1 above)
Comment (umaar) [reply lib/client/metrics/metrics.js]: Was more just flagging as a potential thing, if at all, as there are ` new prometheusClient.Histogram()` things happening within a getter, but maybe that's fine!

File: package.json (hunk 2 above)
Comment (solidgoldpig) [reply package.json]: NB. this was occasionally biting me and taking me too long for the penny to drop that I thought it sensible to enforce it

File: lib/client/metrics/metrics.js (hunk 1 above)
Comment (solidgoldpig) [reply lib/client/metrics/metrics.js]: If you fancy adding a test that proves that the meat of the init function only ever gets run once, feel free ;)

**Pull request merged**